src/TestProblem/Hydro/ClusterMerger/Init_TestProb_ClusterMerger.cpp

```diff
@@ -57,24 +68,96 @@ static FieldIdx_t ColorField1Idx = Idx_Undefined;
+       int    JetDirection_case;          // methods for choosing the jet direction
+                                          //    1: fixed at x-axis;
+                                          //    2: import from table (generate JetDirection.txt)
+                                          //    3: align with angular momentum
+       bool   fixBH;                      // fix the BH at the simulation box center and set its velocity to be zero (1 cluster only)
 // =======================================================================================
 
 
 // problem-specific function prototypes
 #ifdef MASSIVE_PARTICLES
-long Read_Particle_Number_ClusterMerger(std::string filename);
-void Par_Init_ByFunction_ClusterMerger(const long NPar_ThisRank,
-                                       const long NPar_AllRank,
-                                       real_par *ParMass, real_par *ParPosX, real_par *ParPosY, real_par *ParPosZ,
-                                       real_par *ParVelX, real_par *ParVelY, real_par *ParVelZ, real_par *ParTime,
-                                       real_par *ParType, real_par *AllAttribute[PAR_NATT_TOTAL]);
+long Read_Particle_Number_ClusterMerger( std::string filename );
+void Par_Init_ByFunction_ClusterMerger( const long NPar_ThisRank,
+                                        const long NPar_AllRank,
+                                        real_par *ParMass, real_par *ParPosX, real_par *ParPosY, real_par *ParPosZ,
+                                        real_par *ParVelX, real_par *ParVelY, real_par *ParVelZ, real_par *ParTime,
+                                        real_par *ParType, real_par *AllAttribute[PAR_NATT_TOTAL] );
+void Aux_Record_ClusterMerger();
 #endif
 
-int Read_Num_Points_ClusterMerger(std::string filename);
-void Read_Profile_ClusterMerger(std::string filename, std::string fieldname,
-                                double field[]);
+bool Flag_ClusterMerger( const int i, const int j, const int k, const int lv, const int PID, const double *Threshold );
+int Read_Num_Points_ClusterMerger( std::string filename );
+void Read_Profile_ClusterMerger( std::string filename, std::string fieldname, double field[] );
 void AddNewField_ClusterMerger();
 
+int Flu_ResetByUser_Func_ClusterMerger( real fluid[], const double Emag, const double x, const double y, const double z,
+                                        const double Time, const double dt, const int lv, double AuxArray[] );
+void Flu_ResetByUser_API_ClusterMerger( const int lv, const int FluSg, const int MagSg, const double TimeNew, const double dt );
+
+extern void (*Flu_ResetByUser_API_Ptr)( const int lv, const int FluSg, const int MagSg, const double TimeNew, const double dt );
+void Output_ClusterMerger();
+void Init_User_ClusterMerger();
+static herr_t LoadField( const char *FieldName, void *FieldPtr, const hid_t H5_SetID_Target,
+                         const hid_t H5_TypeID_Target );
```

Comment:
- [ ] Make sure that the code can still be compiled when disabling HDF5.
- [ ] Raise an error when running this test problem without HDF5.